### ISSUE TITLE
Run simulation button is missing

### ISSUE THREAD

**Issue by minhduc0711** [gama-platform/gama#3090]
**Describe the bug**
The green play button, along with the step button and cycle time slider do not show up after running an experiment.
I skimmed the console logs but there was no exception at all, but only these:
```
Adding object pool: Envelope 3D
Creation of Game of Life0
Closing msi.gama.application.view.InteractiveConsoleView
```
If I reload the experiment, the play button will appear, but clicking on it does not do anything. This exception also appears:
```
Closing msi.gama.application.view.ConsoleView
Creation of Game of Life0
java.lang.IllegalStateException: Resolver already set!
        at com.jogamp.common.util.JarUtil.setResolver(JarUtil.java:92)
        at ummisco.gama.opengl.view.OpenGLInitializer.run(OpenGLInitializer.java:36)
        at ummisco.gama.ui.utils.SwtGui.prepareForExperiment(SwtGui.java:455)
        at msi.gama.kernel.experiment.ExperimentPlan.open(ExperimentPlan.java:506)
        at msi.gama.kernel.experiment.ExperimentPlan.open(ExperimentPlan.java:517)
        at msi.gama.kernel.experiment.ExperimentPlan.reload(ExperimentPlan.java:532)
        at msi.gama.kernel.experiment.ExperimentController.processUserCommand(ExperimentController.java:129)
        at msi.gama.kernel.experiment.ExperimentController.run(ExperimentController.java:65)
        at java.base/java.lang.Thread.run(Thread.java:832)
```

**To Reproduce**
1. Open any GAML model
2. Run an experiment (play button is missing)
3. Reload the experiment
4. Click on the play button (which is non-functional)

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/34294427/114806280-af9c1e00-9dce-11eb-8079-d22e9ebdb950.png)

**Desktop (please complete the following information):**
 - OS: Linux 5.11.13-arch1-1
 - PC Model: Acer Aspire A5
 - GAMA version: 1.8.2 Alpha
 - Java version: Embedded?
 - Graphics cards / Display system: Intel iGPU + Wayland

**Comment by minhduc0711** [comment]
I resolved this by reverting this merge [`833cf6d` (#3069)](https://github.com/gama-platform/gama/pull/3069/commits/833cf6dfad99d9af8f6d2c795f6c9014a94a1e5c) on my branch.

This is the first time I pull from `GAMA_1.8.2` since when I created my branch, so the last working commit is from quite a long time ago https://github.com/gama-platform/gama/commit/ec9bb7cb7a826a2fff2ef3e2129b4aeb962825d9.

**Comment by AlexisDrogoul** [comment]
I cant reproduce this on macOS, and I guess it has something to do with the error on OpenGL you are displaying (the `prepareForExperiment()` method probably never finishes, leaving the UI in a precarious state). 
I suspect lines 54 to 61 of `OpenGLInitializer` to be the cause of this behavior on your specific configuration. For whatever reason, `OpenGLProfile` may not report that it is initialised, entering in an infinite loop and preventing `isInitialized` to become true and creating an error the second time you enter the method. 

Could you try to comment lines 55 to 61 and retry it ? Thanks !

**Comment by minhduc0711** [comment]
Thanks Alexis, but it did not work unfortunately. 

I should mention that I can reproduce this bug even with the model `msi.gama.models/models/GAML\ Syntax/Conditional\ Structures/When.gaml`, so perhaps it is not an OpenGL issue.

**Comment by AlexisDrogoul** [comment]
Same trace with or without the removal of the lines ? The OpenGL initialisation takes place anyway whenever a GUI experiment is launched (whether or not OpenGL is used).

**Comment by minhduc0711** [comment]
I meant that the behavior is still the same after commenting out these lines.

**Comment by AlexisDrogoul** [comment]
So it is probably the new version of JOGL that sports some initialisation errors on Wayland... @RoiArthurB, do you have the same issue ? Problem is that I cant spot any exception coming from the initialiser on macOS. 

One way to proceed is 
- to get the sources of JOGL here : https://jogamp.org/deployment/jogamp-next/fat/
https://jogamp.org/deployment/jogamp-next/fat/jogamp-fat-java-src.zip 
- link these sources with the libraries, for example trying to access to the class `com.jogamp.opengl.GLProfile`
- put JOGL on debug mode by setting some properties (mostly `-Djogl.debug=all` in the arguments of gama, I think, but the class above provides some hints on which properties to set).
- run various experiments and see what log is produced each time... 

Thanks !

**Comment by AlexisDrogoul** [comment]
Just tested the `Djogl.debug=all` in the runtime product and it seems to produce the right trace. 

<img width="692" alt="Eclipse Workspace for GAMA 1… 2021-04-16 08-55-30" src="https://user-images.githubusercontent.com/579256/114960610-9ad48e80-9e91-11eb-9439-2c428885865b.png">

**Comment by minhduc0711** [comment]
> So it is probably the new version of JOGL that sports some initialisation errors on Wayland.
I can reproduce this issue on X as well. I think @ndgnuh also saw this on his X-running machine.

**Comment by minhduc0711** [comment]
@AlexisDrogoul
Here is the exception I got, it seems to appear only once when I first run an experiment
```
GLProfile.initSingleton() - thread Thread-12
    [2]: com.jogamp.opengl.GLProfile.initSingleton(GLProfile.java:216)
    [3]: ummisco.gama.opengl.view.OpenGLInitializer.run(OpenGLInitializer.java:54)
    [4]: ummisco.gama.ui.utils.SwtGui.prepareForExperiment(SwtGui.java:455)
    [5]: msi.gama.kernel.experiment.ExperimentPlan.open(ExperimentPlan.java:506)
    [6]: msi.gama.kernel.experiment.ExperimentPlan.open(ExperimentPlan.java:517)
    [7]: msi.gama.kernel.experiment.ExperimentController.lambda$0(ExperimentController.java:94)
    [8]: java.base/java.lang.Thread.run(Thread.java:834)
```

The relevant lines in `GLProfile.java`:
```
if(DEBUG) {
    System.err.println("GLProfile.initSingleton() - thread "+Thread.currentThread().getName());
    ExceptionUtils.dumpStack(System.err);  // line 216
}

```

**Comment by AlexisDrogoul** [comment]
Well. Not very useful as it is not an exception ... I'm afraid you will have to play with the different options offered by GLProfile for initialising OpenGL. See for instance the ones exposed in the `static` section of the class. 

```
    static {
        // Also initializes TempJarCache if shall be used.
        Platform.initSingleton();
        final boolean isIOS = Platform.OSType.IOS == Platform.getOSType();
        final boolean isOSXorIOS = Platform.OSType.MACOS == Platform.getOSType() || isIOS;

        DEBUG = Debug.debug("GLProfile");
        disabledEGL = isIOS;
        disableOpenGLCore = PropertyAccess.isPropertyDefined("jogl.disable.openglcore", true) && !isOSXorIOS;
        disableOpenGLARBContext = PropertyAccess.isPropertyDefined("jogl.disable.openglarbcontext", true) && !isOSXorIOS;
        disableOpenGLES = disableOpenGLARBContext || PropertyAccess.isPropertyDefined("jogl.disable.opengles", true);
        disableOpenGLDesktop = PropertyAccess.isPropertyDefined("jogl.disable.opengldesktop", true);
        disableSurfacelessContext = PropertyAccess.isPropertyDefined("jogl.disable.surfacelesscontext", true);
        enableANGLE = PropertyAccess.isPropertyDefined("jogl.enable.ANGLE", true);
    }
```

Finally, which JDK are you using ? A quick look at some of the posts in http://forum.jogamp.org/jogl-f782158.html seems to indicate that the ones provided by AdoptOpenJDK work better under the zillions of versions of Linux ...

**Comment by minhduc0711** [comment]
Sorry, these logs were printed in red so my caveman brain assumed it was an exception...

I'm using OpenJDK-11, will try AdoptOpenJDK to see if it helps.

**Comment by ptaillandier** [comment]
And with Oracle JDK?

**Comment by ndgnuh** [comment]
I get the error even with the bundled JDK, so it was not the JDK

**Comment by AlexisDrogoul** [comment]
@ndgnuh is the bundled JDK using AdoptOpenJDK ?

**Comment by ndgnuh** [comment]
> @ndgnuh is the bundled JDK using AdoptOpenJDK ?

To clarify, the bundled JDK is the JDK that comes with Gama (On the release page)

**Comment by AlexisDrogoul** [comment]
In any case, we need to raise an issue in JOGL about this. But we need to be more specific about what the problem is in the initialiser, for example by activating / deactivating some specific options (see the code above about the arguments that can be passed ...). If we find a combination of arguments that allow OpenGL to be initialized, then we can expect them to issue a fix quickly...

**Comment by hqnghi88** [comment]
sorry, may be not related with the issue, but do you, linux guys, have a working application with 3D environment and if it is a java application , it should help to understand some case

**Comment by ndgnuh** [comment]
Can you name an application or two? I'll try. AFAIK, libreoffice was working very well, but IDK if it has 3D environment (maybe not).

**Comment by hqnghi88** [comment]
No, i dont ask for a new challenge, i ask for experience with such kind of 3d application/3d display, 3d visualization on your linux distribution.

**Comment by minhduc0711** [comment]
@hqnghi88 Does the working commit ec9bb7c count? :eyes:

**Comment by AlexisDrogoul** [comment]
Best thing would be to run the tests / executables that can be found on the Jogamp site. Can anyone try this ?

**Comment by minhduc0711** [comment]
I tried running [this test](https://jogamp.org/wiki/index.php?title=Jogamp_Versioning_and_Releases#Runtime_Version_Check), hope it is the one you were referring to.

I noticed that I got different errors with two JDK versions:
1. OpenJDK 11 ([full log](https://pastebin.com/raw/GgBZTcwq)): 
```
Inconsistency detected by ld.so: dl-lookup.c: 105: check_match: Assertion `version->filename == NULL || ! _dl_name_match_p (version->filename, map)' failed!
```
2. AdoptOpenJDK 11 ([full log](https://pastebin.com/raw/sGMMgtUc)):
```
Exception in thread "main" java.lang.RuntimeException: com.jogamp.opengl.GLException: Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f11a80
17790, owner true, ResourceToolkitLock[obj 0x33abfb37, isOwner true, <68145517, 12a6abe7>[count 1, qsz 0, owner <main-Display-.x11_:0-1-EDT-1>]]], but: [GLProfile[GLES1/GLES1.hw], GLProfile[
GLES2/GLES3.hw], GLProfile[GL2ES1/GLES1.hw], GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GLES3/GLES3.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw
], GLProfile[GL2GL3/GL4.hw]]
        at jogamp.newt.DefaultEDTUtil.invokeImpl(DefaultEDTUtil.java:252)
        at jogamp.newt.DefaultEDTUtil.invoke(DefaultEDTUtil.java:165)
        at jogamp.newt.DisplayImpl.runOnEDTIfAvail(DisplayImpl.java:442)
        at jogamp.newt.WindowImpl.runOnEDTIfAvail(WindowImpl.java:2782)
        at jogamp.newt.WindowImpl.setVisible(WindowImpl.java:1330)
        at jogamp.newt.WindowImpl.setVisible(WindowImpl.java:1335)
        at com.jogamp.newt.opengl.GLWindow.setVisible(GLWindow.java:578)
        at com.jogamp.newt.opengl.GLWindow.main(GLWindow.java:1127)
Caused by: com.jogamp.opengl.GLException: Profile GL4bc is not available on X11GraphicsDevice[type .x11, connection :0, unitID 0, handle 0x7f11a8017790, owner true, ResourceToolkitLock[obj 0
x33abfb37, isOwner true, <68145517, 12a6abe7>[count 1, qsz 0, owner <main-Display-.x11_:0-1-EDT-1>]]], but: [GLProfile[GLES1/GLES1.hw], GLProfile[GLES2/GLES3.hw], GLProfile[GL2ES1/GLES1.hw],
 GLProfile[GL4ES3/GL4.hw], GLProfile[GL2ES2/GL4.hw], GLProfile[GL4/GL4.hw], GLProfile[GLES3/GLES3.hw], GLProfile[GL4/GL4.hw], GLProfile[GL3/GL4.hw], GLProfile[GL2GL3/GL4.hw]]
        at com.jogamp.opengl.GLProfile.get(GLProfile.java:991)
        at jogamp.opengl.GLContextImpl.verifyInstance(GLContextImpl.java:1471)
        at jogamp.opengl.GLContextImpl.setGLFunctionAvailability(GLContextImpl.java:1942)
        at jogamp.opengl.x11.glx.X11GLXContext.createImpl(X11GLXContext.java:395)
        at jogamp.opengl.GLContextImpl.makeCurrentWithinLock(GLContextImpl.java:765)
        at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:648)
        at jogamp.opengl.GLContextImpl.makeCurrent(GLContextImpl.java:586)
        at jogamp.opengl.GLDrawableHelper.invokeGLImpl(GLDrawableHelper.java:1279)
        at jogamp.opengl.GLDrawableHelper.invokeGL(GLDrawableHelper.java:1147)
        at com.jogamp.newt.opengl.GLWindow.display(GLWindow.java:759)
        at jogamp.opengl.GLAutoDrawableBase.defaultWindowResizedOp(GLAutoDrawableBase.java:260)
        at com.jogamp.newt.opengl.GLWindow.access$200(GLWindow.java:119)
        at com.jogamp.newt.opengl.GLWindow$2.windowResized(GLWindow.java:141)
        at jogamp.newt.WindowImpl.consumeWindowEvent(WindowImpl.java:4383)
        at jogamp.newt.WindowImpl.sendWindowEvent(WindowImpl.java:4317)
        at jogamp.newt.WindowImpl.setVisibleActionImpl(WindowImpl.java:1306)
        at jogamp.newt.WindowImpl$VisibleAction.run(WindowImpl.java:1318)
        at com.jogamp.common.util.RunnableTask.run(RunnableTask.java:145)
        at jogamp.newt.DefaultEDTUtil$NEDT.run(DefaultEDTUtil.java:375)
```
Hope this helps!

**Comment by AlexisDrogoul** [comment]
Well. Thanks ! It means JOGL has some problems with your version of Linux. Or conversely. The first error is already documented on JogAmp forums, I think. The second, well... I don't know. Are your graphic drivers up to date ?

**Comment by minhduc0711** [comment]
I believe this is a related issue: https://github.com/processing/processing/issues/5476

I tried one workaround in the comments, which is setting `System.setProperty("jogl.disable.openglcore", "false");` before the `GLProfile.initSingleton()`. Did not work still.

**Comment by AlexisDrogoul** [comment]
Probably related to another issue, #3033, already solved but maybe reappearing on another form. The problem is that the versions of JOGL are a bit fuzzy... 

@minhduc0711 by the way, you might want to pass the preferences from the launch configuration instead of setting them in the code, as I don't know when GLProfile is loaded.

**Comment by minhduc0711** [comment]
Like this right?
```
-Djogl.disable.openglcore=true
```
It still doesn't seem to work..

**Comment by AlexisDrogoul** [comment]
@minhduc0711 

1. Reading your logs, it seems that you ran the tests of JOGL using v. 2.3.2. Actually, the one to use is 2.4.0RC (the latest one, which can be retrieved from here : https://jogamp.org/deployment/autobuilds/master/?C=M;O=D). 
2. If you obtain new exceptions, would it be possible for you to post an issue in the JogAmp forum dedicated to JOGL (http://forum.jogamp.org/jogl-f782158.html) and to report them here ? 

Thanks !

**Comment by minhduc0711** [comment]
I tried running 2.4.0rc, and there seems to be no error strangely: [logs](https://pastebin.com/raw/R3GiybbW).

**Comment by ndgnuh** [comment]
I tried to compile Gama on the latest version of JOGL and the latest commit of Gama_1.8.2. Experiment shows nothing until I hit reload, on Java2D it still shows nothing, but the experiment will run, on OpenGL display exception was thrown:

```
com.jogamp.opengl.GLException: No default device available
	at com.jogamp.opengl.GLProfile.getProfileMap(GLProfile.java:2327)
	at com.jogamp.opengl.GLProfile.get(GLProfile.java:1009)
	at com.jogamp.opengl.GLProfile.getDefault(GLProfile.java:743)
	at com.jogamp.opengl.GLProfile.getDefault(GLProfile.java:754)
	at ummisco.gama.opengl.view.SWTOpenGLDisplaySurface.createCanvas(SWTOpenGLDisplaySurface.java:135)
	at ummisco.gama.opengl.view.SWTOpenGLDisplaySurface.createAnimator(SWTOpenGLDisplaySurface.java:130)
	at ummisco.gama.opengl.view.SWTOpenGLDisplaySurface.<init>(SWTOpenGLDisplaySurface.java:116)
	at gaml.additions.opengl.GamlAdditions.lambda$9(GamlAdditions.java:73)
	at msi.gama.common.interfaces.IDisplayCreator$DisplayDescription.create(IDisplayCreator.java:37)
	at msi.gama.common.interfaces.IDisplayCreator$DisplayDescription.create(IDisplayCreator.java:47)
	at ummisco.gama.ui.utils.SwtGui.getDisplaySurfaceFor(SwtGui.java:266)
	at ummisco.gama.opengl.view.OpenGLDisplayView.createSurfaceComposite(OpenGLDisplayView.java:49)
	at ummisco.gama.ui.views.displays.LayeredDisplayView.ownCreatePartControl(LayeredDisplayView.java:154)
	at ummisco.gama.ui.views.GamaViewPart.createPartControl(GamaViewPart.java:200)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.createPartControl(CompatibilityPart.java:158)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityView.createPartControl(CompatibilityView.java:155)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.create(CompatibilityPart.java:365)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:999)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:964)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalInject(InjectorImpl.java:140)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:405)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:332)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:202)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:91)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:60)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:42)
	at org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer.createWidget(ContributedPartRenderer.java:132)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createWidget(PartRenderingEngine.java:999)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:659)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$1.run(PartRenderingEngine.java:544)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:528)
	at org.eclipse.e4.ui.workbench.renderers.swt.ElementReferenceRenderer.createWidget(ElementReferenceRenderer.java:73)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createWidget(PartRenderingEngine.java:999)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:659)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:765)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:730)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:714)
	at org.eclipse.e4.ui.workbench.renderers.swt.StackRenderer.showTab(StackRenderer.java:1209)
	at org.eclipse.e4.ui.workbench.renderers.swt.LazyStackRenderer.postProcess(LazyStackRenderer.java:114)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:677)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:765)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:730)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:714)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:71)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashRenderer.processContents(SashRenderer.java:150)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:673)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:765)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:730)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:714)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.subscribeTopicToBeRendered(PartRenderingEngine.java:161)
	at jdk.internal.reflect.GeneratedMethodAccessor96.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.di.internal.extensions.EventObjectSupplier$DIEventHandler.handleEvent(EventObjectSupplier.java:92)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:205)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:203)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:133)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:75)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:44)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:55)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:63)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:424)
	at org.eclipse.e4.ui.model.application.ui.impl.UIElementImpl.setToBeRendered(UIElementImpl.java:314)
	at org.eclipse.e4.ui.workbench.addons.cleanupaddon.CleanupAddon.subscribeRenderingChanged(CleanupAddon.java:334)
	at jdk.internal.reflect.GeneratedMethodAccessor95.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.swt.widgets.Synchronizer.syncExec(Synchronizer.java:236)
	at org.eclipse.ui.internal.UISynchronizer.syncExec(UISynchronizer.java:133)
	at org.eclipse.swt.widgets.Display.syncExec(Display.java:5941)
	at org.eclipse.e4.ui.workbench.swt.DisplayUISynchronize.syncExec(DisplayUISynchronize.java:34)
	at org.eclipse.e4.ui.internal.di.UIEventObjectSupplier$UIEventHandler.handleEvent(UIEventObjectSupplier.java:64)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:205)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:203)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:133)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:75)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:44)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:55)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:63)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:424)
	at org.eclipse.e4.ui.model.application.ui.impl.UIElementImpl.setToBeRendered(UIElementImpl.java:314)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.showElementInWindow(ModelServiceImpl.java:655)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.showElementInWindow(ModelServiceImpl.java:662)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.bringToTop(ModelServiceImpl.java:624)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.delegateBringToTop(PartServiceImpl.java:786)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:757)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:679)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.activate(PartServiceImpl.java:674)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.showPart(PartServiceImpl.java:1251)
	at org.eclipse.ui.internal.WorkbenchPage.showPart(WorkbenchPage.java:1252)
	at org.eclipse.ui.internal.WorkbenchPage.busyShowView(WorkbenchPage.java:1240)
	at org.eclipse.ui.internal.WorkbenchPage.lambda$11(WorkbenchPage.java:4249)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:74)
	at org.eclipse.ui.internal.WorkbenchPage.showView(WorkbenchPage.java:4247)
	at ummisco.gama.ui.utils.SwtGui.lambda$0(SwtGui.java:200)
	at ummisco.gama.ui.utils.WorkbenchHelper.run(WorkbenchHelper.java:115)
	at ummisco.gama.ui.utils.SwtGui.internalShowView(SwtGui.java:191)
	at ummisco.gama.ui.utils.SwtGui.showView(SwtGui.java:230)
	at msi.gama.outputs.AbstractDisplayOutput.lambda$0(AbstractDisplayOutput.java:40)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)
	at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:5067)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4573)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at msi.gama.application.Application.start(Application.java:143)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1461)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1434)
```

**Comment by AlexisDrogoul** [comment]
@ndgnuh This basically means that OpenGL has not been properly initialised. Why, I don't know. 

@minhduc0711 @ndgnuh are you able to run the demos proposed by JOGL and display them ?

**Comment by minhduc0711** [comment]
These demos work fine for me https://github.com/sgothel/jogl-demos.
I could not figure out how to run the demos available in [the link you provided](https://jogamp.org/deployment/autobuilds/master/?C=M;O=D) though.

**Comment by minhduc0711** [comment]
Hi @AlexisDrogoul, I'm planning to merge my branch into 1.8.2 soon, but then I realize it would be hard for me to continue working on the skill with this issue in the way...
Switching to another OpenGL library will probably take a lot of time, so I'm willing to find a workaround with JOGL for now. Could you suggest where to start?

**Comment by AlexisDrogoul** [comment]
No idea for the moment. I guess the first thing would be to try to reproduce the problem *without* GAMA and submit it to the JOGL forum/issues.

**Comment by RoiArthurB** [comment]
The fix doesn't work in the GAMA 1.8.2 w/ JDK new build on Linux (Seems to be a library missing) : 

```
-- Surface bounds set to 0  0 | 1339 871
org.eclipse.swt.SWTError: Unspecified error [Error while starting AWT] (java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	/home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/configuration/org.eclipse.osgi/161/0/.cp/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/configuration/org.eclipse.osgi/161/0/.cp/libswt-awt-gtk-4942r22.so)
	no swt-awt-gtk in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	/home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so)
	Can't load library: /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk.so
	/home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so)
)
	at org.eclipse.swt.SWT.error(SWT.java:4915)
	at org.eclipse.swt.awt.SWT_AWT.new_Frame(SWT_AWT.java:166)
	at ummisco.gama.java2d.swing.SwingControl.populate(SwingControl.java:71)
	at ummisco.gama.java2d.swing.SwingControl.setBounds(SwingControl.java:104)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:983)
	at org.eclipse.swt.custom.SashFormLayout.layout(SashFormLayout.java:110)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:231)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:231)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:216)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:983)
	at org.eclipse.swt.custom.CTabFolderLayout.layout(CTabFolderLayout.java:119)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:983)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.setRectangle(SashLayout.java:315)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.tileSubNodes(SashLayout.java:260)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.tileSubNodes(SashLayout.java:303)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.tileSubNodes(SashLayout.java:303)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.tileSubNodes(SashLayout.java:303)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashLayout.layout(SashLayout.java:178)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.setLayoutDeferred(Composite.java:1694)
	at org.eclipse.swt.widgets.Display.runDeferredLayouts(Display.java:5116)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4557)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at msi.gama.application.Application.start(Application.java:143)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1461)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1434)
Caused by: java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	/home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/configuration/org.eclipse.osgi/161/0/.cp/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/configuration/org.eclipse.osgi/161/0/.cp/libswt-awt-gtk-4942r22.so)
	no swt-awt-gtk in java.library.path: /usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	/home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so)
	Can't load library: /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk.so
	/home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so: /home/roiarthurb/.local/share/GAMA_1.8.2_Linux_with_JDK/jdk/lib/libjawt.so: version `SUNWprivate_1.1' not found (required by /home/roiarthurb/.swt/lib/linux/x86_64/libswt-awt-gtk-4942r22.so)

	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:338)
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:257)
	at org.eclipse.swt.awt.SWT_AWT.loadLibrary(SWT_AWT.java:77)
	at org.eclipse.swt.awt.SWT_AWT.new_Frame(SWT_AWT.java:162)
	... 79 more
-- Surface bounds set to 0  0 | 1339 889
-- Surface bounds set to 0  0 | 1339 889
java.lang.UnsatisfiedLinkError: 'java.lang.Object org.eclipse.swt.awt.SWT_AWT.initFrame(long, java.lang.String)'
	at org.eclipse.swt.awt.SWT_AWT.initFrame(Native Method)
	at org.eclipse.swt.awt.SWT_AWT.new_Frame(SWT_AWT.java:170)
	at ummisco.gama.java2d.swing.SwingControl.populate(SwingControl.java:71)
	at ummisco.gama.java2d.swing.SwingControl.setBounds(SwingControl.java:104)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:983)
	at org.eclipse.swt.custom.SashFormLayout.layout(SashFormLayout.java:110)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:697)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:201)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:231)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:231)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.setBounds(Composite.java:1635)
	at org.eclipse.swt.widgets.Control.setBounds(Control.java:1023)
	at org.eclipse.swt.layout.FillLayout.layout(FillLayout.java:216)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1851)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.updateLayout(Composite.java:1857)
	at org.eclipse.swt.widgets.Composite.setLayoutDeferred(Composite.java:1694)
	at org.eclipse.swt.widgets.Display.runDeferredLayouts(Display.java:5116)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4557)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)
	at msi.gama.application.Application.start(Application.java:143)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1461)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1434)
Tree root = horizontal weight null
-- Surface bounds set to 0  0 | 667 871
-- Surface bounds set to 0  0 | 667 871
-- Surface bounds set to 0  0 | 667 889
-- Surface bounds set to 0  0 | 667 889
```

**Comment by AlexisDrogoul** [comment]
Seems like a completely different issue (SWR requiring some AWR libraries. You dont have this problem when you open a Java2D display ?

**Comment by RoiArthurB** [comment]
I'll migrate the issue, actually it happens only on `java2D` and graphs displays...
